src/HDF5.jl

```diff
@@ -535,25 +536,33 @@ heuristic_chunk(x) = Int[]
 
 ### High-level interface ###
 # Open or create an HDF5 file
-function h5open(filename::AbstractString, rd::Bool, wr::Bool, cr::Bool, tr::Bool, ff::Bool,
+function h5open(filename::AbstractString, mode::String,
```

Comment:
I was reluctant to document the methods more fully because I feel like the pv argument should be refactored to accept `Pair`s instead of relying on alternation of keys and values.